test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/UserTests.kt

```diff
@@ -353,6 +381,118 @@ class UserTests {
         }
     }
 
+    @Test
+    fun linkCredentials_emailPassword() = runBlocking {
+        val anonUser = app.login(Credentials.anonymous())
+        assertEquals(1, anonUser.identities.size)
+        val (email, password) = randomEmail() to "123456"
+        app.emailPasswordAuth.registerUser(email, password)
+        val linkedUser = anonUser.linkCredentials(Credentials.emailPassword(email, password))
+
+        assertSame(anonUser, linkedUser)
+        assertEquals(2, linkedUser.identities.size)
+        assertEquals(AuthenticationProvider.EMAIL_PASSWORD, linkedUser.identities[1].provider)
+
+        // Validate that we cannot link a second set of credentials
+        val otherEmail = randomEmail()
+        val otherPassword = "123456"
+        app.emailPasswordAuth.registerUser(otherEmail, otherPassword)
+        val credentials = Credentials.emailPassword(otherEmail, otherPassword)
+
+        assertFailsWith<CredentialsCannotBeLinkedException> {
```

Comment:
That method doesn't support running suspend functions in the closure and there doesn't seem to be an easy fix for it as it requires rewriting some of the stdlib functions we depend on. It would be a nice future improvement though.